pkg/metadata/metadata.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Re-export metadata in an opaque format.
+package metadata
+
+import (
+	"github.com/pulumi/pulumi-terraform-bridge/v3/internal/metadata"
+)
+
+// A store persisted between `tfgen` and a running provider.
+type Provider *metadata.Data
```

Comment:
Could we name this ProviderMetadata? A bit more verbose but much less confusing. This sounds like it's a Provider that provides services about metadata.

Reply:
Sure. Because `metadata.ProviderMetadata` is overly verbose, and because all functions in `metadata` also need to be renamed to match, I folded `metadata` into `tfbridge`. The new name is `tfbridge.NewProviderMetadata(...)`.